chalk/core.py

```diff
@@ -270,8 +283,61 @@ def accept(self, visitor: DiagramVisitor[A], **kwargs: Any) -> A:
 class ApplyName(BaseDiagram):
     """ApplyName class."""
 
-    dname: str
+    dname: Name
     diagram: Diagram
 
     def accept(self, visitor: DiagramVisitor[A], **kwargs: Any) -> A:
         return visitor.visit_apply_name(self, **kwargs)
+
+
+class Qualify(DiagramVisitor[Diagram]):
```

Comment:
Interesting. I was thinking we could do this by having a "ApplyQualifier" node, that gets checked by the other visitor. Changing the tree scares me a little bit.

Reply:
Okay, I can add a new `ApplyQualifier` variant to our abstract syntax tree (AST). I think my approach was a reflex to keep the core AST minimal, but indeed traversing the AST when running `qualify` might incur too much computational burden.

Reply:
I trust your judgement.